libtsuba/src/RDGPartHeader.h

```diff
@@ -271,7 +338,38 @@ class KATANA_EXPORT RDGPartHeader {
   /// Metadata filled in by CuSP, or from storage (meta partition file)
   PartitionMetadata metadata_;
 
+  // tracks changes to json on disk structure of the PartitionHeader
+  // current one is defined by latest_storage_format_version_
+  uint32_t storage_format_version_;
```

Comment:
@tylershunt just an FYI for eventual review. 

I added `storage_format_version` so that we can track json layout changes to the RDGPartHeader. 
This made it easy to differentiate between part headers that should have the new EntityTypeID structures vs ones that don't in the RDGPartHeader `from_json`, and the RDG `Make` and `Validate` funcitons
defaulted to using `1` for old version, `2` for new.
on store, we always write the latest

Reply:
Sounds good.